server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/NewSizeMeasuresStep.java

```diff
@@ -141,11 +172,30 @@ public void initialize(CounterInitializationContext context) {
       }
 
       Map<Period, Integer> newLinesDuplicatedByPeriod = duplicationCounters.getNewLinesDuplicated();
-      context.getPeriods().forEach(period -> {
-        newLines.increment(period, newLinesDuplicatedByPeriod.getOrDefault(period, 0));
-        newBlocks.increment(period, duplicationCounters.getNewBlocksDuplicated().getOrDefault(period, 0));
+      periods.forEach(period -> {
+        newDuplicatedLines.increment(period, newLinesDuplicatedByPeriod.getOrDefault(period, 0));
+        newDuplicatedBlocks.increment(period, duplicationCounters.getNewBlocksDuplicated().getOrDefault(period, 0));
       });
     }
+
+    /**
+     * NCLOC_DATA contains Key-value pairs, where key - is a line number, and value - is an indicator of whether line
+     * contains code (1) or not (0).
+     *
+     * This method parses the value of the NCLOC_DATA measure and return the line numbers which contain code.
+     */
+    private static List<Integer> nclocLineNumbers(Measure nclocDataMeasure) {
+      Map<Integer, Integer> parsedNclocData = KeyValueFormat.parse(nclocDataMeasure.getData(), newIntegerConverter(), newIntegerConverter());
+      return parsedNclocData.entrySet()
+        .stream()
+        .filter(entry -> entry.getValue() == 1)
+        .map(Map.Entry::getKey)
+        .collect(Collectors.toList());
+    }
+
+    private static boolean isLineInPeriod(Changeset changeset, Period period) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "isLineInPeriod" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod)